templates/wrappers.hs

```diff
@@ -30,24 +30,30 @@ import qualified Data.Bits
 
 -- | Encode a Haskell String to a list of Word8 values, in UTF8 format.
 utf8Encode :: Char -> [Word8]
-utf8Encode = map fromIntegral . go . ord
+utf8Encode = uncurry (:) . utf8Encode'
+
+utf8Encode' :: Char -> (Word8, [Word8])
+utf8Encode' = map' fromIntegral . go . ord
```

Comment:
`utf8Encode' c = (fromIntregral x, map fromIntegral xs) where !(x,xs) = go (ord c)`

and then get rid of `map'`

(not sure if we can use `BangPatterns` here, but if not you can do `case go (ord c) of (x,xs) -> ...`)